### PR TITLE
COM-297: Refactor theming of `CopyToClipboardButton`

### PR DESCRIPTION
**Pull request by stekalt**
<details>
    <summary>CopyToClipboardButton</summary>

https://github.com/vivid-planet/comet/assets/71731085/049cccc2-4c97-4d43-a426-9fef891da0c3

</details>

<details>
    <summary>Themed CopyToClipboardButton</summary>

comet-copyToClipboardButton-themed

</details>

### CODE REVIEW

**Review by jamesricky — CHANGES_REQUESTED** [review]

File: packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx [hunk 1]
```diff
@@ -1,47 +1,53 @@
...
+    const [showSuccess, setShowSuccess] = React.useState<boolean | undefined>(false);
 
-    const [showSuccess, setShowSuccess] = React.useState<boolean>(false);
+    const ownerState: OwnerState = {
+        showSuccess: showSuccess,
+    };
 
     const copyTextToClipboard = () => {
-        navigator.clipboard.writeText(copyText);
+        navigator.clipboard.writeText(inProps.copyText);
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx:50]: Please only use values from the theme-props, never the `inProps`

File: packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx [hunk 2]
```diff
@@ -1,47 +1,53 @@
...
-    successButton?: Partial<IconButtonProps>;
-    copyIcon?: Partial<SvgIconProps>;
-    successIcon?: Partial<SvgIconProps>;
-}
-
-export interface CopyToClipboardButtonProps {
+export interface CopyToClipboardButtonProps
+    extends ThemedComponentBaseProps<{
+        root: "div";
+        buttonContainer: "div";
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx:9]: There is no slot named `buttonContainer` so this is not necessary

```suggestion
```

File: packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx [hunk 3]
```diff
@@ -1,47 +1,53 @@
...
+        copyButtonContainer: "div";
+        successButtonContainer: "div";
+        button: "div";
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx:12]: There is no slot named button so this is not necessary

```suggestion
```

File: packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx [hunk 4]
```diff
@@ -1,47 +1,53 @@
...
-        copyIcon: copyIconProps,
-        successIcon: successIconProps,
-    } = componentsProps;
+        copyIcon = <Copy />,
+        successIcon = <Accept />,
+        slotProps,
+        ...restProps
+    } = useThemeProps({ props: inProps, name: "CometAdminCopyToClipboardButton" });
+
+    const [showSuccess, setShowSuccess] = React.useState<boolean | undefined>(false);
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx:43]: I don't think this should ever be `undefined`

```suggestion
    const [showSuccess, setShowSuccess] = React.useState<boolean>(false);
```

File: packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx [hunk 5]
```diff
@@ -1,47 +1,53 @@
...
+    | "showSuccess"
+    | "buttonContainer"
+    | "copyButtonContainer"
+    | "successButtonContainer"
+    | "button"
+    | "copyButton"
+    | "successButton";
 
+type OwnerState = {
+    showSuccess: boolean | undefined;
```
Comment (jamesricky) [inline packages/admin/admin/src/common/buttons/CopyToClipboardButton.tsx:32]: I don't think this should ever be `undefined`

```suggestion
    showSuccess: boolean;
```

**Review by jamesricky — CHANGES_REQUESTED** [review]

**Review by jamesricky — APPROVED** [review]

**Review by johnnyomair — APPROVED** [review]

**Pull request merged**